### PR TITLE
Make servlet context path key private

### PR DESCRIPTION
**Pull request by laurit**
Addresses a review comment from https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4078

### CODE REVIEW

File: instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/BaseServletHelper.java [hunk 1]
```diff
@@ -67,12 +67,8 @@ protected Context customizeContext(Context context, REQUEST request) {
...
   }
 
-  private Context addServletContextPath(Context context, REQUEST request) {
-    String contextPath = accessor.getRequestContextPath(request);
-    if (contextPath != null && !contextPath.isEmpty() && !contextPath.equals("/")) {
-      return context.with(ServletContextPath.CONTEXT_KEY, contextPath);
-    }
-    return context;
+  protected Context addServletContextPath(Context context, REQUEST request) {
+    return ServletContextPath.init(context, () -> accessor.getRequestContextPath(request));
```
Comment (trask) [inline instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/BaseServletHelper.java:71]: can we do non-capturing lambda here similar to #4014?

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/servlet/ServletContextPath.java [hunk 2]
```diff
@@ -22,18 +23,41 @@
 
   // Keeps track of the servlet context path that needs to be prepended to the route when updating
   // the span name
-  public static final ContextKey<String> CONTEXT_KEY =
+  private static final ContextKey<ServletContextPath> CONTEXT_KEY =
```
Comment (trask) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/servlet/ServletContextPath.java:26]: do we need the `ServletContextPath` wrapper object?

File: instrumentation/servlet/servlet-common/library/src/main/java/io/opentelemetry/instrumentation/servlet/ServletHttpServerTracer.java [hunk 3]
```diff
@@ -42,9 +43,11 @@
       Config.get().getBoolean("otel.instrumentation.servlet.experimental-span-attributes", false);
 
   private final ServletAccessor<REQUEST, RESPONSE> accessor;
+  private final Function<REQUEST, String> contextPathExtractor;
 
   protected ServletHttpServerTracer(ServletAccessor<REQUEST, RESPONSE> accessor) {
     this.accessor = accessor;
+    this.contextPathExtractor = (request) -> accessor.getRequestContextPath(request);
```
Comment (mateuszrzeszutek) [inline instrumentation/servlet/servlet-common/library/src/main/java/io/opentelemetry/instrumentation/servlet/ServletHttpServerTracer.java:50]: Nit
```suggestion
    this.contextPathExtractor = accessor::getRequestContextPath;
```

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/servlet/ServletContextPath.java (hunk 2 above)
Comment (laurit) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/servlet/ServletContextPath.java:26]: Using wrapper helps to distinguish context path not yet set from context path empty. I guess I could use `""` to mark empty context path.

**Review by trask — APPROVED** [review]

**Pull request merged**